helm/generic-node/templates/archive-node.yaml

```diff
@@ -0,0 +1,171 @@
+---
+{{if .Values.archive.enabled}}
+apiVersion: apps/v1
+{{- if .Values.features.persistence.enabled}}
```

Comment:
Archive node is always a Deployment, the archive itself does not care what database it talks to. Postgres is the statefulset in this case.